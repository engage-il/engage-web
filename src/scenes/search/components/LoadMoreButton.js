// @flow
import React, { Component } from 'react'
import { css } from 'glamor'
import type { Rule } from 'glamor'
import { stylesheet, colors, mixins } from 'shared/styles'

export class LoadMoreButton extends Component {
  props: {
    hasMore: boolean,
    onClick: Function,
    styles?: Rule
  }

  // events
  didClickLink = () => {
    this.props.onClick()
  }

  // lifecycle
  render () {
    const { styles, hasMore } = this.props
    if (!hasMore) {
      return null
    }

    return <a {...css(rules.button, styles)} onClick={this.didClickLink}>
      Load More
    </a>
  }
}

const rules = stylesheet({
  button: {
    ...mixins.shadows.make(colors.primaryShadow, 5),
    height: 40,
    minWidth: 100,
    padding: '0 5%',
    fontSize: 18,
    lineHeight: '40px',
    textAlign: 'center',
    color: colors.white,
    backgroundColor: colors.primary,
    transition: 'background-color 0.25s',
    ':hover': {
      backgroundColor: colors.primaryHighlight
    }
  }
})
