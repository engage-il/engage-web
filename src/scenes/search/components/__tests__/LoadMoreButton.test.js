/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { LoadMoreButton } from '../LoadMoreButton'

// subject
let subject
let hasMore
let onClick

function loadSubject () {
  subject = shallow(<LoadMoreButton hasMore={hasMore} onClick={onClick} />)
}

const element = {
  link: () => subject.find('a')
}

// specs
beforeEach(() => {
  hasMore = true
  onClick = jest.fn()
})

afterEach(() => {
  subject = null
})

describe('#render', () => {
  describe('when there is more to load', () => {
    beforeEach(loadSubject)

    it('renders the button', () => {
      expect(element.link()).toBePresent()
    })
  })

  describe('when has more is false', () => {
    beforeEach(() => {
      hasMore = false
      loadSubject()
    })

    it('hides the button', () => {
      expect(subject.children()).toBeEmpty()
    })
  })
})

describe('on click', () => {
  beforeEach(() => {
    loadSubject()
    element.link().prop('onClick')()
  })

  it('notifies its parent', () => {
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
