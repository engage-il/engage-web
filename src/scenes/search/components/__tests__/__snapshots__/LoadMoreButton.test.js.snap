// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render on click 1`] = `
<a
  data-css-1kk31r9=""
  onClick={[Function]}
>
  Load More
</a>
`;

exports[`#render when has more is false 1`] = `null`;

exports[`#render when there is more to load 1`] = `
<a
  data-css-1kk31r9=""
  onClick={[Function]}
>
  Load More
</a>
`;
